### PR TITLE
SONARJAVA-4224: Enable SpringBeansShouldBeAccessibleCheck to use incremental analysis with caching

### PR DESCRIPTION
**Pull request by johann-beleites-sonarsource**

### CODE REVIEW

File: java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java [hunk 1]
```diff
@@ -92,17 +129,108 @@ void testSpringBootApplicationWithAnnotation() {
...
+    verify(check, times(15)).visitNode(any());
+    verify(check, times(2)).scanWithoutParsing(any());
+    assertThat(writeCache.getData())
+      .hasSize(2);
+
+
+    check = spy(new SpringBeansShouldBeAccessibleCheck());
+
+    var populatedReadCache = new InternalReadCache().putAll(writeCache);
+    var writeCache2 = new InternalWriteCache().bind(populatedReadCache);
```
Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java:174]: Rename this to `endCache`

Comment (dorian-burihabwa-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java:174]: Rename this to `endCache`

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
LGTM 👍🏿 A change to the log level on cache misses would help with debugging

File: java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java (hunk 1 above)
Comment (johann-beleites-sonarsource) [reply java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java:174]: How about `finalWriteCache`?

Comment (dorian-burihabwa-sonarsource) [reply java-checks/src/test/java/org/sonar/java/checks/spring/SpringBeansShouldBeAccessibleCheckTest.java:174]: Sounds good

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
Looking good and ready to merge

**Pull request merged**